### PR TITLE
Moved Pattern Matching Test from FSharpQA to ComponentTests

### PR DESCRIPTION
**Pull request by edgarfgp**
- [x] Check that all assertions are correct
- [x] Remove the old tests once the team and CI are happy

This will help with https://github.com/dotnet/fsharp/pull/15733 as I can run the pattern matching tests easily

### CODE REVIEW

**Comment by KevinRansom** [comment]
Nice!!!!!!

File: tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // This test was automatically generated (moved from FSharpQA suite - Conformance/PatternMatching/And)
+    [<Theory; Directory(__SOURCE_DIRECTORY__, Includes=[|"andPattern03.fs"|])>]
+    let ``And - andPattern03_fs - --test:ErrorRanges`` compilation =
+        compilation
+        |> asFs
+        |> withOptions ["--test:ErrorRanges"]
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 41, Line 7, Col 28, Line 7, Col 45, "A unique overload for method 'TryParse' could not be determined based on type information prior to this program point. A type annotation may be needed.
```
Comment (KevinRansom) [inline tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: pretty sure this should be: shouldSucceed

From env.lst:
SOURCE=andPattern03.fs	# andPattern03.fs

From: andPattern03.fs
````
// #Conformance #PatternMatching #PatternMatchingGuards 
#light

open System

let (|ToInt|) x =
    let (parsed, result) = Int32.TryParse(x)
    if parsed then result
    else           -1

let (|ToStr|) (x : int) = x.ToString()

let test input =
    match input with
    | ToInt (ToStr "1")           -> 1
    | ToInt 2 & ToInt (ToStr "2") -> 2
    | ToInt (ToStr "3") & ToInt (ToStr (ToInt (ToStr "3"))) -> 3
    | _ -> -1

if test "1" <> 1 then exit 1
if test "2" <> 2 then exit 2
if test "3" <> 3 then exit 3

exit 0
````

Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: Yeah that is what I expected, but `Int32.TryParse(x)` was throwing a compiler error that I might need to ignore maybe ?

Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: I will look into this tomorrow :). Thanks for the review

Comment (smoothdeveloper) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: @edgarfgp likely an added overload in recent dotnet releases, maybe it is just fixing `x: string` in the signature that is needed?

Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: Yep, precisely what's happening.

Comment (KevinRansom) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: My only question about that is why does it succeed in fsharpqa

Comment (KevinRansom) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs:37]: Hmm!!!!

````
c:\kevinransom\fsharp>cd \temp

c:\temp>fsc c:\kevinransom\fsharp\tests\fsharpqa\Source\Conformance\PatternMatching\And\andPattern03.fs
Microsoft (R) F# Compiler version 12.8.0.0 for F# 8.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

c:\temp>
````

Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/And.fs]: Ok, this is passing now as in`fsharpqa`. Will continue checking the rest of the assertions

**Comment by psfinaki** [comment]
Thanks Edgar, thanks for investing in our testing life :)
Hopefully git will handle moved files so that we can review it more or less thoroughly.

**Comment by edgarfgp** [comment]
> Thanks Edgar, thanks for investing in our testing life :) Hopefully git will handle moved files so that we can review it more or less thoroughly.

Removed the old test. Let's see how it handles the moved files.

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**